test/unit/server/handlers/consents/{ID}/revoke.test.ts

```diff
@@ -0,0 +1,256 @@
+          .toBeCalledWith(completeConsentRevoked, request)
+        expect(mockRevokeConsentStatus).toBeCalledWith(completeConsentRevoked)
+        expect(mockGeneratePatchConsentRequest)
+          .toBeCalledWith(completeConsentRevoked)
+        expect(mockPatchConsents)
+          .toBeCalledWith(consentId,
+            requestBody,
+            request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
+          )
+      })
+
+    it('Should throw an error due to consent retrieval error',
+      async (): Promise<void> => {
+        mockConsentRetrieve.mockRejectedValueOnce(new Error('Test'))
+        await expect(Handler.validateRequestAndRevokeConsent(request))
+          .rejects.toThrowError('NotImplementedYetError')
+
+        expect(mockConsentRetrieve).toBeCalledWith(consentId)
+        expect(mockIsConsentRequestValid).not.toBeCalled()
+        expect(mockRevokeConsentStatus).not.toBeCalled()
+        expect(mockGeneratePatchConsentRequest).not.toBeCalled()
+        expect(mockPatchConsents).not.toBeCalled()
+      })
+
+    it('Should throw an error due to invalid request',
+      async (): Promise<void> => {
+        mockIsConsentRequestValid.mockReturnValueOnce(false)
+        await expect(Handler.validateRequestAndRevokeConsent(request))
+          .rejects.toThrowError('NotImplementedYetError')
+
+        expect(mockConsentRetrieve).toBeCalledWith(consentId)
+        expect(mockIsConsentRequestValid).toBeCalledWith(partialConsentActive)
+        expect(mockRevokeConsentStatus).not.toBeCalled()
+        expect(mockGeneratePatchConsentRequest).not.toBeCalled()
+        expect(mockPatchConsents).not.toBeCalled()
+      })
+
+    it('Should throw an error as patchConsents() throws an error',
+      async (): Promise<void> => {
+        mockPatchConsents.mockRejectedValue(new Error('Test Error'))
+        await expect(Handler.validateRequestAndRevokeConsent(request))
+          .rejects.toThrowError('Test Error')
+
+        expect(mockConsentRetrieve).toBeCalledWith(consentId)
+        expect(mockIsConsentRequestValid)
+          .toBeCalledWith(completeConsentRevoked, request)
+        expect(mockRevokeConsentStatus).not.toBeCalled()
+        expect(mockGeneratePatchConsentRequest)
+          .toBeCalledWith(completeConsentRevoked)
+        expect(mockPatchConsents)
+          .toBeCalledWith(consentId,
+            requestBody,
+            request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
+          )
+      }
+    )
+  })
+
+  describe('Post', (): void => {
+    const mockValidateRequestAndRevokeConsent = jest.spyOn(
+      Handler, 'validateRequestAndRevokeConsent')
+
+    beforeAll((): void => {
+      mockValidateRequestAndRevokeConsent.mockResolvedValue()
+    })
+
+    it('Should return 202 success code',
+      async (): Promise<void> => {
+        const response = await Handler.post(
+          request as Request,
+          h as ResponseToolkit
+        )
+        expect(response).toBe(202)
```

Comment:
this is false positive test -> @akapur99 tell me why?

Reply:
wasn't really sure what to test in the main handler - just need to confirm that as long background function is called - handler resolves

Reply:
@akapur99 NOPE! it is false positive because the mock in L80 is invalid